### PR TITLE
Rework Token API comments

### PR DESCRIPTION
**Pull request by ianw**
Move the token API discussion into a common section discussing the
generation and listing of the tokens.  Add a note on the display of
the sha1 during creation and listing.

### CODE REVIEW

File: docs/content/doc/developers/api-usage.en-us.md [hunk 1]
```diff
@@ -40,8 +40,42 @@ better understand this by looking at the code -- as of this writing,
...
+an additional header that contains the one time password (6 digit
+rotating token). An example of the header is `X-Gitea-OTP: 123456`
+where `123456` is where you'd place the code from your
+authenticator. Here is how the request would look like in curl:
+
+```sh
+$ curl -H "X-Gitea-OTP: 123456" --request GET --url https://yourusername:yourpassword@gitea.your.host/api/v1/users/yourusername/tokens
+```
+
+You can also create an API key token via your Gitea installation's web
```
Comment (noerw) [inline docs/content/doc/developers/api-usage.en-us.md:77]: I'd put this first in this section as it is very short, and doesn't get lost in the API noise ;)

File: docs/content/doc/developers/api-usage.en-us.md [hunk 2]
```diff
@@ -40,8 +40,42 @@ better understand this by looking at the code -- as of this writing,
...
+The ``sha1`` (the token) is only returned once and is not stored in
+plain-text.  It will not be displayed when listing tokens with a `GET`
+request; e.g.
+
+```sh
+$ curl --request GET --url https://yourusername:password@gitea.your.host/api/v1/users/<username>/tokens
+[{"name":"test","sha1":"","token_last_eight:"........":},{"name":"dev","sha1":"","token_last_eight":"........"}]
+```
+
+As of v1.8.0 of Gitea, if using basic authentication with the API and
```
Comment (noerw) [inline docs/content/doc/developers/api-usage.en-us.md]: Reword so it's clearer that this is only for 2FA. And we don't need to hint at changes from 1.8, released in 2019 anymore ;)
```suggestion
To use the API with basic authentication with two factor authentication
enabled, you'll need to send an additional header that contains the one
time password (6 digitrotating token).
An example of the header is `X-Gitea-OTP: 123456` where `123456`
is where you'd place the code from your authenticator.
Here is how the request would look like in curl:
```

**Review by noerw — CHANGES_REQUESTED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@ianw pleace resolve conflicts

**Comment by noerw** [comment]
@6543 that github code review suggestion thing messed things up (how do your reference multiple lines to be replaced in that UI anyway?)

**Review by noerw — APPROVED** [review]

**Pull request merged**